include/secp256k1_frost.h

```diff
@@ -0,0 +1,643 @@
+ *  In:      sig: pointer to the signature
+ */
+SECP256K1_API int secp256k1_frost_partial_sig_serialize(
+    const secp256k1_context* ctx,
+    unsigned char *out32,
+    const secp256k1_frost_partial_sig* sig
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Parse a FROST partial signature.
+ *
+ *  Returns: 1 when the signature could be parsed, 0 otherwise.
+ *  Args:    ctx: a secp256k1 context object
+ *  Out:     sig: pointer to a signature object
+ *  In:     in32: pointer to the 32-byte signature to be parsed
+ *
+ *  After the call, sig will always be initialized. If parsing failed or the
+ *  encoded numbers are out of range, signature verification with it is
+ *  guaranteed to fail for every message and public key.
+ */
+SECP256K1_API int secp256k1_frost_partial_sig_parse(
+    const secp256k1_context* ctx,
+    secp256k1_frost_partial_sig* sig,
+    const unsigned char *in32
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Creates a key generation share and verifiable secret sharing ("VSS")
+ *  commitments.
+ *
+ *  The share is derived deterministically from the input parameters. TODO:
+ *  explain session_id, general usage, etc.
+ *
+ *  TODO: explicitly state the requirements on the communication channel, i.e.
+ *  the shares are supposed to be sent through secure (authenticated and
+ *  encrypted) channels to the other signers. Also, the vss_hashes must be
+ *  authentic, etc.
+ *
+ *  Returns: 0 if the arguments are invalid, 1 otherwise
+ *  Args:            ctx: pointer to a context object initialized for
+ *                        verification
+ *  Out:  vss_commitment: the coefficient commitments. The length of this array
+ *                        must be equal to the threshold (can be NULL).
+ *                 share: pointer to the polynomial share
+ *   In:    session_id32: a 32-byte session_id32 as explained above
+ *               keypair: pointer to a keypair that contains the secret that is
+ *                        shared
+ *                    pk: pointer to the public key of the share recipient
+ *             threshold: the minimum number of signers required to produce a
+ *                        signature
+ */
+SECP256K1_API int secp256k1_frost_share_gen(
+    const secp256k1_context *ctx,
+    secp256k1_pubkey *vss_commitment,
+    secp256k1_frost_share *share,
+    const unsigned char *session_id32,
+    const secp256k1_keypair *keypair,
+    const secp256k1_xonly_pubkey *pk,
+    size_t threshold
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4) SECP256K1_ARG_NONNULL(5) SECP256K1_ARG_NONNULL(6);
+
+/** Aggregates shares
+ *
+ *  As part of the key generation protocol, each participant receives a share
+ *  from each participant, including a share they "receive" from themselves.
+ *  This function verifies those shares against their VSS commitments,
```

Comment:
Note to reviewers: I think this part has been actually split out in `secp256k1_frost_share_verify`.